### ISSUE TITLE
Version 2.6.0 does not compile in Chef 11.16.4

### ISSUE THREAD

**Issue by martinisoft** [sensu/sensu-chef#330]
I cannot run the 2.6.0 version of the Sensu cookbook on Chef 11.16.4 with Ubuntu 14.04.2. It appears the new default attribute change is not working?

Here is a stack trace:

```
Compiling Cookbooks...

================================================================================
Recipe Compile Error in /var/chef/cache/cookbooks/sensu/attributes/default.rb
================================================================================

ArgumentError
-------------
wrong number of arguments (0 for 1)

Cookbook Trace:
---------------
  /var/chef/cache/cookbooks/sensu/attributes/default.rb:48:in `from_file'

Relevant File Content:
----------------------
/var/chef/cache/cookbooks/sensu/attributes/default.rb:

 41:
 42:  # api
 43:  default.sensu.api.host = "localhost"
 44:  default.sensu.api.bind = "0.0.0.0"
 45:  default.sensu.api.port = 4567
 46:
 47:  # data bag
 48>> default.sensu.data_bag.name = "sensu"
 49:  default.sensu.data_bag.ssl_item = "ssl"
 50:  default.sensu.data_bag.config_item = "config"
 51:  default.sensu.data_bag.enterprise_item = "enterprise"
 52:


Running handlers:
Running handlers complete
Chef Client failed. 0 resources updated in 12.67055051 seconds
```

**Comment by damacus** [comment]
I'm also getting this exact error with chef-12.1.2-1

I've created a dummy config bag ( which wasn't needed before) and that doesn't seem to have fixed it.

**Comment by martinisoft** [comment]
Still seeing this error on Chef 12.3 as well. Seems https://github.com/sensu/sensu-chef/commit/2a1e852c578384da7b3c0882892cdd548b7d2e90 introduced this breaking change.

**Comment by damacus** [comment]
Looks like data_bag_key was switched for data_bag_item

**Comment by martinisoft** [comment]
@damacus Think he cleaned that up here: https://github.com/sensu/sensu-chef/commit/7e9f9c8da644c60a6b480856c16ae51d90615a76

Still seems to be breaking with the dot notation there. Don't know why.

**Comment by damacus** [comment]
Hmm, annoying. I was getting the same error. But am not using sensu
anymore.

My "work around"  was to pin the entire environment to 2.5.

**Comment by martinisoft** [comment]
@damacus That's what I have done. I'm just wondering why only we seem to be the ones experiencing this issue. Want to get to the bottom of it since we're that far behind master.

**Comment by martinisoft** [comment]
Looks like this method of access is colliding with Chef internals, specifically [this method](https://github.com/chef/chef/blob/97677f911e85b2b0db4247723930f30fb4a02c66/lib/chef/dsl/data_query.rb#L47) and causing a problem. 

Was able to trace it with `DEBUG: ["/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.16.4/lib/chef/dsl/data_query.rb", 47]` which was generated by doing `Chef::Log.debug(node.sensu.method(:data_bag).source_location)` in the attribute file.

So, I can propose a PR for @portertech to change the attribute access to bracket notation instead of method so it works more consistently.

**Comment by cwjohnston** [comment]
Closing as this issue seems to be resolved by #350
